### PR TITLE
Py3 changes only

### PR DESCRIPTION
**Pull request by jaredmixpanel**

### CODE REVIEW

File: __init__.py [hunk 1]
```diff
@@ -991,15 +989,15 @@ def export_events(self, output_file, params, format='json', timezone_offset=None
         if self.timeout == 120:
             self.timeout = 1200
 
+        date_format = '%Y-%m-%d'
```
Comment (RyanBuccellato) [inline __init__.py:992]: does moving this out accomplish anything?

File: __init__.py [hunk 2]
```diff
@@ -1021,6 +1019,8 @@ def export_events(self, output_file, params, format='json', timezone_offset=None
                     current_file = current_file + '.gz'
                 with open(current_file, 'wb') as fp:
                     shutil.copyfileobj(events, fp, buffer_size)
+                if not add_gzip_header and compress:
```
Comment (RyanBuccellato) [inline __init__.py:1022]: not entirely sure what the gzip portion is but I would have assumed that the gzip header would take us here. Confirming that only when there is _no_ header we want to run this `_gzip_file` fn

File: __init__.py [hunk 3]
```diff
@@ -1350,7 +1352,10 @@ def _properties_from_csv_row(row, header, ignored_columns):
             else:
                 try:
                     # We use literal_eval() here to de-stringify numbers, lists and objects in the CSV data
-                    p = literal_eval(row[x])
+                    # p = literal_eval(row[x])
```
Comment (RyanBuccellato) [inline __init__.py:1355]: don't need this anymore? Can we just remove the comments?

File: __init__.py [hunk 4]
```diff
@@ -1350,7 +1352,10 @@ def _properties_from_csv_row(row, header, ignored_columns):
...
+                    p = json.loads(row[x])
+                    if isinstance(p, list):
+                        Mixpanel.LOGGER.debug(str(p))
```
Comment (RyanBuccellato) [inline __init__.py:1358]: do we need this debug statement?

File: __init__.py [hunk 5]
```diff
@@ -1720,7 +1730,7 @@ def _send_batch(self, base_url, endpoint, batch, dataset_id=None, dataset_versio
 
         """
         try:
-            params = {'data': base64.b64encode(json.dumps(batch))}
+            params = {'data': base64.b64encode(json.dumps(batch).encode('utf-8'))}
```
Comment (RyanBuccellato) [inline __init__.py:1733]: don't you have to like `.decode` it as well or something iirc? See this thread when I was helping Cat w/ py3 https://mixpanel.slack.com/archives/C03QFM9BL/p1614279849203600?thread_ts=1614278061.196100&cid=C03QFM9BL

**Review by RyanBuccellato — APPROVED** [review]
LGTM

File: __init__.py (hunk 1 above)
Comment (jaredmixpanel) [reply __init__.py:992]: yea it was inside of an if statement and then `date_format` was referenced later in another if statement that had the exact same condition so moved out of the  if statement

File: __init__.py (hunk 2 above)
Comment (jaredmixpanel) [reply __init__.py:1022]: right this is the case where the request was made without the gzip header but they wanted the output compressed to a gzip file.

File: __init__.py (hunk 3 above)
Comment (jaredmixpanel) [reply __init__.py:1355]: removed in later a commit on the full PR

File: __init__.py (hunk 4 above)
Comment (jaredmixpanel) [reply __init__.py:1358]: removed in later a commit on the full PR

File: __init__.py (hunk 5 above)
Comment (jaredmixpanel) [reply __init__.py:1733]: fixed in later a commit on the full PR

**Pull request merged**